app/src/app_machine_probe.c

```diff
@@ -37,6 +37,17 @@ LOG_MODULE_REGISTER(app_machine_probe, LOG_LEVEL_DBG);
 #define SHT30_INIT_TIME K_MSEC(100)
 #define SHT30_CONV_TIME K_MSEC(100)
 
+/* Machine Probe SHT sensor type selection
+ * Configure via CONFIG_APP_MACHINE_PROBE_SHT43 in prj.conf
+ */
+#if defined(CONFIG_APP_MACHINE_PROBE_SHT43)
```

Comment:
I don't like this, as this leads to multiple firmware versions again. We should solve this on a runtime level.